src/man/kproplog.8
.TH "KPROPLOG" "8" "January 06, 2012" "0.0.1" "MIT Kerberos"
.SH NAME
kproplog \- display the contents of the Kerberos principal update log
.
.nr rst2man-indent-level 0
.
.de1 rstReportMargin
\\$1 \\n[an-margin]
level \\n[rst2man-indent-level]
level margin: \\n[rst2man-indent\\n[rst2man-indent-level]]
-
\\n[rst2man-indent0]
\\n[rst2man-indent1]
\\n[rst2man-indent2]
..
.de1 INDENT
.\" .rstReportMargin pre:
. RS \\$1
. nr rst2man-indent\\n[rst2man-indent-level] \\n[an-margin]
. nr rst2man-indent-level +1
.\" .rstReportMargin post:
..
.de UNINDENT
. RE
.\" indent \\n[an-margin]
.\" old: \\n[rst2man-indent\\n[rst2man-indent-level]]
.nr rst2man-indent-level -1
.\" new: \\n[rst2man-indent\\n[rst2man-indent-level]]
.in \\n[rst2man-indent\\n[rst2man-indent-level]]u
..
.\" Man page generated from reStructeredText.
.
.SH SYNOPSIS
.sp
\fBkproplog\fP [\fB\-h\fP] [\fB\-e\fP \fInum\fP] [\-v]
.SH DESCRIPTION
.sp
The \fIkproplog\fP command displays the contents of the Kerberos principal
update log to standard output.
It can be used to keep track of the \fIincremental\fP updates
to the principal database, when enabled.
The update log file contains the update log maintained by the \fIkadmind\fP process
on the master KDC server and the \fIkpropd\fP process on the slave KDC servers.
When updates occur, they are logged to this file.
Subsequently any KDC slave configured for \fIincremental\fP updates will request
the current data from the master KDC and update their \fIprincipal.ulog\fP file
with any updates returned.
.sp
The \fIkproplog\fP command can only be run on a KDC server by someone with privileges
comparable to the superuser.
It will display update entries for that server only.
.sp
If no options are specified, the summary of the update log is displayed.
If invoked on the master, all of the update entries are also displayed.
When invoked on a slave KDC server, only a summary of the updates are displayed,
which includes the serial number of the last update received and
the associated time stamp of the last update.
.SH OPTIONS
.INDENT 0.0
.INDENT 3.5
.INDENT 0.0
.TP
.B \fB\-h\fP
.sp
Display a summary of the update log.
This information includes the database version number, state of the database,
the number of updates in the log, the time stamp of the first and last update,
and the version number of the first and last update entry.
.TP
.B \fB\-e\fP \fInum\fP
.sp
Display the last \fInum\fP update entries in the log.
This is useful when debugging synchronization between KDC servers.
.TP
.B \fB\-v\fP
.sp
Display individual attributes per update.  An example of the output generated for one entry:
.sp
.nf
.ft C
Update Entry
   Update serial # : 4
   Update operation : Add
   Update principal : test@EXAMPLE.COM
   Update size : 424
   Update committed : True
   Update time stamp : Fri Feb 20 23:37:42 2004
   Attributes changed : 6
         Principal
         Key data
         Password last changed
         Modifying principal
         Modification time
         TL data
.ft P
.fi
.UNINDENT
.UNINDENT
.UNINDENT
.SH ENVIRONMENT
.sp
\fIkproplog\fP uses the following environment variables:
.INDENT 0.0
.INDENT 3.5
.INDENT 0.0
.IP \(bu 2
.
\fBKRB5_KDC_PROFILE\fP
.UNINDENT
.UNINDENT
.UNINDENT
.SH SEE ALSO
.sp
kpropd(8)
.SH AUTHOR
MIT
.SH COPYRIGHT
2011, MIT
.\" Generated by docutils manpage writer.
.


src/slave/kproplog.M
.\" slave/kprop.M
.\"
.\" Copyright 2008 by the Massachusetts Institute of Technology.
.\"
.\" Export of this software from the United States of America may
.\"   require a specific license from the United States Government.
.\"   It is the responsibility of any person or organization contemplating
.\"   export to obtain such a license before exporting.
.\" 
.\" WITHIN THAT CONSTRAINT, permission to use, copy, modify, and
.\" distribute this software and its documentation for any purpose and
.\" without fee is hereby granted, provided that the above copyright
.\" notice appear in all copies and that both that copyright notice and
.\" this permission notice appear in supporting documentation, and that
.\" the name of M.I.T. not be used in advertising or publicity pertaining
.\" to distribution of the software without specific, written prior
.\" permission.  Furthermore if you modify this software you must label
.\" your software as modified software and not distribute it in such a
.\" fashion that it might be confused with the original M.I.T. software.
.\" M.I.T. makes no representations about the suitability of
.\" this software for any purpose.  It is provided "as is" without express
.\" or implied warranty.
.\" 
.\"
.\" Copyright (c) 2003, Sun Microsystems, Inc.  All Rights Reserved
.\"
.TH KPROPLOG 1
.SH NAME
kproplog \- display the contents of the Kerberos principal update log
.SH SYNOPSIS
.B kproplog
[\fB\-h\fP] [\fB\-e\fP \fInum\fP]
.br
.SH DESCRIPTION
The
.I kproplog
command displays the contents of the Kerberos principal update log to
standard output.  It can be used to keep track of the incremental
updates to the principal database, when enabled.  The
.I /var/krb5/principal.ulog
file contains the update log maintained by the
.I kadmind
process on the master KDC server and the kpropd process on the slave
KDC servers.  When updates occur, they are logged to this
file.  Subsequently any KDC slave configured for incremental updates
will request the current data from the master KDC and update their
.I principal.ulog
file with any updates returned.

The
.I kproplog
command can only be run on a KDC server by someone with privileges
comparable to the superuser.  It will display update entries for that
server only.

If no options are specified, the summary of the update log is
displayed.  If invoked on the master, all of the update entries are
also displayed.  When invoked on a slave KDC server, only a summary of
the updates are displayed, which includes the serial number of the
last update received and the associated time stamp of the last update.

.SH OPTIONS
.TP
\fB\-h\fP
Display a summary of the update log.  This information includes the
database version number, state of the database, the number of updates
in the log, the time stamp of the first and last update, and the
version number of the first and last update entry.
.TP
\fB\-e\fP \fInum\fP
Display the last
.I num
update entries in the log.  This is useful when debugging
synchronization between KDC servers.
.TP
\fB\-v\fP
Display individual attributes per update.
An example of the output generated for one entry:
.nf
 Update Entry
    Update serial # : 4
    Update operation : Add
    Update principal : test@EXAMPLE.COM
    Update size : 424
    Update committed : True
    Update time stamp : Fri Feb 20 23:37:42 2004
    Attributes changed : 6
          Principal
          Key data
          Password last changed
          Modifying principal
          Modification time
          TL data
.fi

.SH FILES
.TP
.I /var/krb5/principal.ulog
The update log file for incremental propagation.

.SH SEE ALSO
kpropd(8)
